subsys/net/openthread/rpc/common/ot_rpc_common.c

```diff
@@ -540,3 +541,119 @@ bool ot_rpc_decode_border_router_config(struct nrf_rpc_cbor_ctx *ctx, otBorderRo
 
 	return true;
 }
+
+bool ot_rpc_encode_dns_query_config(struct nrf_rpc_cbor_ctx *ctx,
+				    const struct otDnsQueryConfig *config)
+{
+	if (config == NULL) {
+		return zcbor_nil_put(ctx->zs, NULL);
+	}
+
+	if (!zcbor_bstr_encode_ptr(ctx->zs, config->mServerSockAddr.mAddress.mFields.m8,
+				   OT_IP6_ADDRESS_SIZE)) {
+		return false;
+	}
+
+	if (!zcbor_uint32_put(ctx->zs, config->mServerSockAddr.mPort)) {
+		return false;
+	}
+
+	if (!zcbor_uint32_put(ctx->zs, config->mResponseTimeout)) {
+		return false;
+	}
+
+	if (!zcbor_uint32_put(ctx->zs, config->mMaxTxAttempts)) {
+		return false;
+	}
+
+	if (!zcbor_uint32_put(ctx->zs, config->mRecursionFlag)) {
+		return false;
+	}
+
+	if (!zcbor_uint32_put(ctx->zs, config->mNat64Mode)) {
+		return false;
+	}
+
+	if (!zcbor_uint32_put(ctx->zs, config->mServiceMode)) {
+		return false;
+	}
+
+	if (!zcbor_uint32_put(ctx->zs, config->mTransportProto)) {
+		return false;
+	}
+
+	return true;
+}
+
+bool ot_rpc_decode_dns_query_config(struct nrf_rpc_cbor_ctx *ctx, bool *decoded_config,
+				    struct otDnsQueryConfig *config)
+{
+	uint32_t tmp;
+	struct zcbor_string bstr;
+
+	if (zcbor_nil_expect(ctx->zs, NULL)) {
```

Comment:
Use `nrf_rpc_is_null` helper. All these functions that check just `ZCBOR_ERR_WRONG_TYPE` are broken.